dali/test/python/test_dali_variable_batch_size.py

```diff
@@ -372,18 +445,294 @@ def pipe_no_input(max_batch_size, input_data, device):
+                 +data,
+                 data * data2,
+                 data + data2,
+                 data - data2,
+                 data / data2,
+                 data // data2,
+                 data ** data2,
+                #  compare_pipelines doesn't work well with bool so promote to int by *
+                 (data == data2) * 1,
+                 (data != data2) * 1,
+                 (data < data2) * 1,
+                 (data <= data2) * 1,
+                 (data > data2) * 1,
+                 (data >= data2) * 1,
+                 data & data,
+                 data | data,
+                 data ^ data,
+                 dmath.abs(data),
+                 dmath.fabs(data),
+                 dmath.floor(data),
+                 dmath.ceil(data),
+                 dmath.pow(data, 2),
+                 dmath.fpow(data, 1.5),
+                 dmath.min(data, 2),
+                 dmath.max(data, 50),
+                 dmath.clamp(data, 10, 50),
+                 dmath.sqrt(data),
+                 dmath.rsqrt(data),
+                 dmath.cbrt(data),
+                 dmath.exp(data),
+                 dmath.log(data),
+                 dmath.log2(data),
+                 dmath.log10(data),
+                 dmath.sin(data),
+                 dmath.cos(data),
+                 dmath.tan(data),
+                 dmath.asin(data),
+                 dmath.acos(data),
+                 dmath.atan(data),
+                 dmath.atan2(data, 3),
+                 dmath.sinh(data),
+                 dmath.cosh(data),
+                 dmath.tanh(data),
+                 dmath.asinh(data),
+                 dmath.acosh(data),
+                 dmath.atanh(data)]
+        pipe.set_outputs(*processed)
+        return pipe
+
+    def get_data(batch_size):
+        test_data_shape = [random.randint(5, 21), random.randint(5, 21), 3]
+        data1 = [np.random.randint(0, 255, size=test_data_shape, dtype=np.uint8) for _ in range(batch_size)]
+        data2 = [np.random.randint(1, 4, size=test_data_shape, dtype=np.uint8) for _ in range(batch_size)]
+        return [data1, data2]
+
+    input_data = [get_data(random.randint(5, 31)) for _ in range(13)]
+    check_pipeline(input_data, pipeline_fn=pipe)
+
+
+def test_squeeze_op():
+    def pipe(max_batch_size, input_data, device, input_layout=None):
+        pipe = Pipeline(batch_size=max_batch_size, num_threads=4, device_id=0)
+        with pipe:
+            # just to drive the variable batch size.
+            data = fn.external_source(source=input_data, cycle=False, device=device, layout=input_layout)
+            out = fn.expand_dims(data, axes=[0, 2], new_axis_names="YZ")
+            out = fn.squeeze(out, axis_names="Z")
+        pipe.set_outputs(out)
+        return pipe
+
+    check_pipeline(generate_data(31, 13, image_like_shape_generator, lo=0, hi=255, dtype=np.uint8),
+                   pipeline_fn=pipe, input_layout="HWC")
+
+
+def test_box_encoder_op():
+    def coco_anchors():
```

Comment:
How about importing it from the coco test rather than having 3 copies of the same piece of code?

Reply:
Done